networklibrary/src/main/java/com/eis/smsnetwork/broadcast/BroadcastReceiver.java

```diff
@@ -0,0 +1,182 @@
+
+import com.eis.communication.network.commands.CommandExecutor;
+import com.eis.smslibrary.SMSManager;
+import com.eis.smsnetwork.RequestType;
+import com.eis.smsnetwork.SMSInvitation;
+import com.eis.smsnetwork.SMSJoinableNetManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+import com.eis.smslibrary.listeners.SMSReceivedServiceListener;
+import com.eis.smsnetwork.SMSNetDictionary;
+import com.eis.smsnetwork.SMSNetSubscriberList;
+import com.eis.smsnetwork.smsnetcommands.SMSAddPeer;
+
+import java.util.Set;
+
+/**
+ * @author Marco Cognolato, Giovanni Velludo, Enrico Cestaro
+ */
+public class BroadcastReceiver extends SMSReceivedServiceListener {
+
+    private static final int REQUEST_FIELD_END_INDEX = 1;
+    public static final String FIELD_SEPARATOR = "¤";
+    // To allow FIELD_SEPARATOR in keys and resources, we escape it with a backslash when
+    // sending those keys and resources through an SMS. Therefore we only split the message
+    // when FIELD_SEPARATOR is not preceded by a backslash.
+    static final String SEPARATOR_REGEX = "(?<!\\\\)" + FIELD_SEPARATOR;
+
+
+    /**
+     * Receives messages from other peers in the network and acts according to the content of those
+     * messages. Messages are composed of different fields, separated by
+     * {@link BroadcastReceiver#FIELD_SEPARATOR}. {@link BroadcastReceiver#FIELD_SEPARATOR}s
+     * preceded by a backslash do not separate fields.
+     * Field 0 contains the {@link RequestType} of the request contained in this message.
+     * The rest of the message varies depending on the {@link RequestType}:
+     * <ul>
+     * <li>Invite: there are no other fields</li>
+     * <li>AcceptInvitation: there are no other fields</li>
+     * <li>AddPeer: fields from 1 to the last one contain the phone numbers of each
+     * {@link com.eis.smslibrary.SMSPeer} we have to add to our network</li>
+     * <li>QuitNetwork: there are no other fields, because this request can only be sent by the
+     * {@link com.eis.smslibrary.SMSPeer} who wants to be removed</li>
+     * <li>AddResource: starting from 1, fields with odd numbers contain keys, their following
+     * (even)
+     * field contains the corresponding value</li>
+     * <li>RemoveResource: fields from 1 to the last one contain the keys to remove</li>
+     * </ul>
+     *
+     * @param message The message passed by {@link com.eis.smslibrary.SMSReceivedBroadcastReceiver}.
+     */
+    @Override
+    public void onMessageReceived(SMSMessage message) {
+        Log.d("BR_RECEIVER", "Message received: " + message.getPeer() + " " + message.getData());
+        String[] fields = message.getData().split(SEPARATOR_REGEX);
+        RequestType request;
+        try {
+            request = RequestType.get(fields[0]);
+        } catch (ArrayIndexOutOfBoundsException | NullPointerException e) {
+            return;
```

Comment:
I think it's a bad style idea to return from a catch block without any sort of exception managing or calling class alerting.

Reply:
The launch of all exceptions caught in this method is caused by the SMS we received not being formatted properly, so there's no need to alert any class, we just received an SMS that seemed to be correct when looking at the first fields, but then turned out to be garbage. Do you think I should call `Log` whenever we receive any such message?

Reply:
I think could be a good idea to call `Log` during dev stages and then, as you said in the PR description, use ProGuard to delete them in release stage and I think that because at least you're aware of what's happening in case some exception is thrown (if it's thrown in an "unexpected"  case).

Reply:
Ok